lib/defaults.js

```diff
@@ -54,12 +54,25 @@ var defaults = {
     return data;
   }],
 
-  transformResponse: [function transformResponse(data) {
+  transformResponse: [function transformResponse(data, headers) {
     /*eslint no-param-reassign:0*/
     if (typeof data === 'string') {
       try {
+        if (data === '') {
+          return {};
+        }
         data = JSON.parse(data);
-      } catch (e) { /* Ignore */ }
+      } catch (e) {
+        if (
+          !utils.isUndefined(headers) &&
+          !utils.isUndefined(headers['content-type']) &&
+          headers['content-type'].match(new RegExp('application/json', 'i'))
```

Comment:
A regular expression is probably unnecessary here. Also, in the current form, it would return a match for `foo-application/json-bar`, which is likely not the intended behavior. If the RegExp is really needed, it should be compiled only once and stored in a variable for better performance.
```suggestion
          headers['content-type'].toLowerCase() === 'application/json'
```

Reply:
What about I change it to:
```suggestion
          headers['content-type'].startsWith('application/json')
```

so it will work for `application/json; charset=utf-8`